test/multiplexed-session.ts

```diff
@@ -0,0 +1,326 @@
+    this.options = options;
+  }
+  async begin(): Promise<void> {}
+}
+
+describe('MultiplexedSession', () => {
+  let multiplexedSession;
+
+  function noop() {}
+  const DATABASE = {
+    createSession: noop,
+    databaseRole: 'parent_role',
+  } as unknown as Database;
+
+  let fakeMuxSession;
+  let createSessionStub;
+  const sandbox = sinon.createSandbox();
+
+  const createSession = (name = 'id', props?): Session => {
+    props = props || {};
+
+    return Object.assign(new Session(DATABASE, name), props, {
+      create: sandbox.stub().resolves(),
+      transaction: sandbox.stub().returns(new FakeTransaction()),
+      getMetadata: sandbox.stub().resolves({multiplexed: true}),
+    });
+  };
+
+  const shouldNotBeCalled = sandbox.stub().throws('Should not be called.');
+
+  beforeEach(() => {
+    fakeMuxSession = createSession();
+    createSessionStub = sandbox
+      .stub(DATABASE, 'createSession')
+      .withArgs({multiplexed: true})
+      .callsFake(() => {
+        return Promise.resolve([fakeMuxSession]);
+      });
+    multiplexedSession = new MultiplexedSession(DATABASE);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('instantiation', () => {
+    it('should localize the database instance', () => {
+      assert.strictEqual(multiplexedSession.database, DATABASE);
+    });
+
+    it('should apply defaults', () => {
+      assert.strictEqual(multiplexedSession.refreshRate, 7);
+    });
+
+    it('should create a session object', () => {
+      assert.deepStrictEqual(multiplexedSession._multiplexedSession, null);
+    });
+
+    it('should inherit from EventEmitter', () => {
+      assert(multiplexedSession instanceof events.EventEmitter);
+    });
+  });
+
+  describe('createSession', () => {
+    let _createSessionStub;
+    let _maintainStub;
+
+    beforeEach(() => {
+      _maintainStub = sandbox.stub(multiplexedSession, '_maintain');
+      _createSessionStub = sandbox
+        .stub(multiplexedSession, '_createSession')
+        .resolves();
+    });
+
+    it('should create mux session', () => {
+      multiplexedSession.createSession();
+      assert.strictEqual(_createSessionStub.callCount, 1);
+    });
+
+    it('should start housekeeping', done => {
+      multiplexedSession.createSession();
+      setImmediate(() => {
+        try {
+          assert.strictEqual(_maintainStub.callCount, 1);
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+    });
+
+    it('should not propagate instance and database not found errors for Multiplexed Session', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        code: Status.NOT_FOUND,
+        message: 'Database not found',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
```

Comment:
We could replace this with 

`assert.doesNotThrow(() => multiplexedSession.createSession());`

Reply:
Updated, thanks for the suggestion!